eventuals/http.h

```diff
@@ -547,9 +566,91 @@ struct _HTTP {
                       continuation.closed_ = true;
 
                       if (!continuation.error_) {
+                        // Build headers map.
+                        std::stringstream headers_buffer_stringstream(
+                            continuation.headers_buffer_.Extract());
+                        std::map<std::string, std::string> headers;
+
+                        // Typical 'headers_buffer_stringstream'
+                        // looks like this:
+                        // --------------------------------
+                        // HTTP/1.1 200
+                        // SomeHeaderKey1: SomeHeaderValue1
+                        // SomeHeaderKey2: SomeHeaderValue2
+                        // --------------------------------
+                        while (!headers_buffer_stringstream.eof()) {
+                          std::string line;
+                          std::getline(headers_buffer_stringstream, line);
+
+                          // Find where ':' is.
+                          auto column_pos = std::find(
+                              line.cbegin(),
+                              line.cend(),
+                              ':');
+
+                          // Eliminate 'HTTP/1.1 200' line.
+                          if (column_pos == line.cend()) {
+                            continue;
+                          }
+
+                          // Assign key and value.
+                          auto key = std::string(line.cbegin(), column_pos);
+                          auto value = std::string(column_pos + 1, line.cend());
+
+                          // Helper lambda for removing leading
+                          // and trailing spaces.
```

Comment:
Let's add a `TODO` here to bring in the [`absl`](https://abseil.io) library which does this all for us. We bring it in for `stout-flags` [here](https://github.com/3rdparty/stout-flags/blob/main/bazel/deps.bzl#L14).